### ISSUE TITLE
HDF5 API docs missing methods

### ISSUE THREAD

**Issue by kylekeppler** [pandas-dev/pandas#31050]
The [user guide IO Tools HDF5 section](https://pandas.io/docs/user_guide/io.html#hdf5-pytables) references a number of methods that are not documented in the [API docs for HDF5](https://pandas.io/docs/reference/io.html#hdfstore-pytables-hdf5).

The following `HDFStore` methods/attributes are mentioned only in the user guide:

Methods:
- `__init__`
- `remove`
- `create_table_index`
- `get_storer`
- `select_as_coordinates`
- `select_as_multiple`
- `append_to_multiple`
- `select_column`
- `flush`

Attributes:
- `is_open`
- `root`

The following are "public" (ie no leading underscore) methods/attributes in 0.25.3 not listed in either location:

Methods:
- `open`
- `close`
- `copy`
- `get_node`
- `items`
- `iteritems`

Attributes:
- `filename`

The "storer object" that is the result of `get_storer` doesn't seem to be documented. This can be a `pandas.io.pytables.FrameFixed` or `pandas.io.pytables.AppendableFrameTable` (and maybe others?). The `nrows` and `table` attributes are referenced in the user guide. There are a number of other attributes that appear public but are not documented. Seems this is starting to get into the implementation weeds, but since a method mentioned in the user guide can return this object, it would be nice to have this explicitly documented.

It would be useful to document the public methods/attributes of `HDFStore` in the API docs. is there any reason these were omitted originally? I think all the would need to be done would be to add to [io.rst](https://github.com/pandas-dev/pandas/blob/master/doc/source/reference/io.rst).

There should probably be a discussion of what should be documented for the storer objects. Perhaps just a reference from the `get_storer` API doc page and a listing of methods/attributes of the abstract base class that are implemented by all subclasses?

**Comment by pavankumar369** [comment]
hey I would like to work on this !

**Comment by zzapzzap** [comment]
ping @pavankumar369 Are you still working on?

**Comment by pavankumar369** [comment]
@zzapzzap  Yep! Kind of busy with exams. Will restart working on it on Tuesday.

**Comment by zzapzzap** [comment]
I see. I hope there will be good results :)
